server/routes/imageUploaderAPI.js

```diff
@@ -1,37 +1,51 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
 
-const multer = require('multer');
-const multerS3 = require('multer-s3');
-const AWS = require('aws-sdk');
+const multer = require("multer");
+const multerS3 = require("multer-s3");
+const AWS = require("aws-sdk");
+
+const auth = require("../middleware/auth");
+const db = require("../db/models");
+const mongoose = require("mongoose");
 
 // Set the region
 AWS.config.update({ region: process.env.AWS_REGION });
 
+const BUCKET = process.env.AWS_BUCKET_NAME;
+
 // Create S3 service object
 const s3 = new AWS.S3();
-const FN = Date.now().toString() + '.png'
+
 // AWS Params put into storage using Multer S3
 const upload = multer({
   storage: multerS3({
     s3: s3,
-    bucket: process.env.AWS_BUCKET_NAME,
+    bucket: BUCKET,
     key: function (req, file, cb) {
+      const FN = Date.now().toString() + ".png";
       cb(null, FN);
     },
   }),
-}).single('file');
+}).single("file");
 
 // Sends Storage Object to S3 Bucket
-router.post('/api/image-upload', (req, res) => {
+router.post("/api/image-upload", auth, (req, res) => {
   upload(req, res, (err) => {
     if (err) {
-      res.status(500).send('There was a server error with the upload');
+      res.status(500).send("There was a server error with the upload");
     } else {
       if (req.file == undefined) {
-        res.status(400).send('There was a error with your upload paramaters');
+        res.status(400).send("There was a error with your upload paramaters");
       } else {
-        res.status(200).send({msg:'Your image was uploaded sucessfully', fileName : req.file.key});
+        const awsUrl = `https://${BUCKET}.s3.amazonaws.com/${req.file.key}`;
+        db.User.updateOne(
+          { _id: req.userId },
+          { $set: { photoUrl: awsUrl } }
+        ).catch((error) => {
+          res.send(error);
+        });
+        res.status(200).send({ awsUrl });
```

Comment:
You'll want to move this to a .then() on Line 45 to solve potential async issuea

Reply:
I did ` .then((res) =>  res.status(200).send({ awsUrl })` originally but caused the app to crash.
So now I've changed it to this: `.then(() => uploadRes.status(200).send({ awsUrl }))` which is working.